src/trix/models/html_parser.js

```diff
@@ -218,13 +218,13 @@ export default class HTMLParser extends BasicObject {
           this.appendAttachmentWithAttributes(attributes, this.getTextAttributes(element))
           return this.processedElements.push(element)
         case "tr":
-          if (element.parentNode.firstChild !== element) {
+          if (this.needsTableSeparator(element)) {
             return this.appendStringWithAttributes("\n")
           }
           break
         case "td":
-          if (element.parentNode.firstChild !== element) {
-            return this.appendStringWithAttributes(" | ")
+          if (this.needsTableSeparator(element)) {
+            return this.appendStringWithAttributes(config.theme.tableSeparator)
```

Comment:
There are two separators at plat here: table row separator (`"\n"`) and table cell separator (`" | "`). I think it'd be better to extract both, for symmetry.